### DISCUSSION TITLE
Pre-baked shadow maps

### DISCUSSION THREAD

**Discussion by Adrian-Samoticha** [bevyengine/bevy#16510] [Q&A]
Hi all,

I’ve been trying to develop a voxel engine (in Godot, but I’m considering moving to Bevy) and noticed that about half of my frame time was dedicated to rendering shadows. Considering the significant performance impact along with the fact that my environments were mostly static, I was wondering if pre-baked shadow maps would provide a solution.

Conceptually, pre-rendered shadows appear quite simple to implement—you simply render the shadow map once and then re-use it for subsequent frames. However, it appears that you cannot prevent Bevy from updating its shadow maps every frame.

Is there something I missed, or do I need to implement my own shadow rendering system to achieve something like this?

**Comment by DGriffin91** [comment]
I think this would be really easy to implement for point and spot lights, directional lights basically follow the camera around so something more complex would be needed in that case.

**Reply by Adrian-Samoticha** [reply]
I think if Bevy allowed the user to control when the shadow maps are updated, you could come up with a solution that fits whatever your needs are.
